diff -Nru xz-5.2.3-old/src/liblzma/Makefile.am xz-5.2.3/src/liblzma/Makefile.am
--- xz-5.2.3-old/src/liblzma/Makefile.am	2015-07-21 14:35:24.124356586 +0000
+++ xz-5.2.3/src/liblzma/Makefile.am	2015-07-21 14:36:25.768356450 +0000
@@ -24,7 +24,7 @@
 	-I$(top_srcdir)/src/liblzma/simple \
 	-I$(top_srcdir)/src/common \
 	-DTUKLIB_SYMBOL_PREFIX=lzma_
-liblzma_la_LDFLAGS = -no-undefined -version-info 7:3:2
+liblzma_la_LDFLAGS = -no-undefined -version-info 7:3:2 -avoid-version
 
 EXTRA_DIST += liblzma.map validate_map.sh
 if COND_SYMVERS
diff -Nru xz-5.2.3-old/src/liblzma/Makefile.in xz-5.2.3/src/liblzma/Makefile.in
--- xz-5.2.3-old/src/liblzma/Makefile.in	2015-07-21 14:35:24.124356586 +0000
+++ xz-5.2.3/src/liblzma/Makefile.in	2015-07-21 14:37:15.412356340 +0000
@@ -750,7 +750,7 @@
 	-I$(top_srcdir)/src/common \
 	-DTUKLIB_SYMBOL_PREFIX=lzma_
 
-liblzma_la_LDFLAGS = -no-undefined -version-info 7:3:2 $(am__append_1) \
+liblzma_la_LDFLAGS = -no-undefined -version-info 7:3:2 -avoid-version $(am__append_1) \
 	$(am__append_42)
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = liblzma.pc

