--- ncurses-6.0/aclocal.m4.orig	2016-05-02 18:32:51.228640173 +0800
+++ ncurses-6.0/aclocal.m4	2016-05-02 18:33:33.248641581 +0800
@@ -5142,7 +5142,6 @@
 AC_CACHE_VAL(cf_cv_prog_$1_c_o,
 [
 cat > conftest.$ac_ext <<CF_EOF
-#include <stdio.h>
 int main()
 {
 	${cf_cv_main_return:-return}(0);
--- ncurses-6.0/configure.orig	2016-05-02 18:44:40.545330548 +0800
+++ ncurses-6.0/configure	2016-05-02 18:44:48.365330810 +0800
@@ -2528,7 +2528,6 @@
 else
 
 cat > conftest.$ac_ext <<CF_EOF
-#include <stdio.h>
 int main()
 {
 	${cf_cv_main_return:-return}(0);
