--- openssl-1.0.2d/Makefile.org.orig	2015-11-28 03:16:14.902907088 +0800
+++ openssl-1.0.2d/Makefile.org	2015-11-28 03:20:12.499581693 +0800
@@ -15,6 +15,13 @@
 CONFIGURE_ARGS=
 SHLIB_TARGET=
 
+ifneq ($(DISABLE_LIBVERSION),1)
+LIBVERSION=$(SHLIB_MAJOR).$(SHLIB_MINOR)
+else
+LIBVERSION=
+endif
+
+
 # HERE indicates where this Makefile lives.  This can be used to indicate
 # where sub-Makefiles are expected to be.  Currently has very limited usage,
 # and should probably not be bothered with at all.
@@ -339,7 +346,7 @@
 link-shared:
 	@ set -e; for i in $(SHLIBDIRS); do \
 		$(MAKE) -f $(HERE)/Makefile.shared -e $(BUILDENV) \
-			LIBNAME=$$i LIBVERSION=$(SHLIB_MAJOR).$(SHLIB_MINOR) \
+			LIBNAME=$$i LIBVERSION=$(LIBVERSION) \
 			LIBCOMPATVERSIONS=";$(SHLIB_VERSION_HISTORY)" \
 			symlink.$(SHLIB_TARGET); \
 		libs="$$libs -l$$i"; \
@@ -353,7 +360,7 @@
 			libs="$(LIBKRB5) $$libs"; \
 		fi; \
 		$(CLEARENV) && $(MAKE) -f Makefile.shared -e $(BUILDENV) \
-			LIBNAME=$$i LIBVERSION=$(SHLIB_MAJOR).$(SHLIB_MINOR) \
+			LIBNAME=$$i LIBVERSION=$(LIBVERSION) \
 			LIBCOMPATVERSIONS=";$(SHLIB_VERSION_HISTORY)" \
 			LIBDEPS="$$libs $(EX_LIBS)" \
 			link_a.$(SHLIB_TARGET); \
