diff -r ad855c779bf3 configure.ac
--- a/configure.ac      Thu Dec 03 17:32:05 2015 -0800
+++ b/configure.ac      Sat Dec 05 13:55:20 2015 +0800
@@ -1800,7 +1827,7 @@
 sys/times.h sys/types.h sys/uio.h sys/un.h sys/utsname.h sys/wait.h pty.h \
 libutil.h sys/resource.h netpacket/packet.h sysexits.h bluetooth.h \
 bluetooth/bluetooth.h linux/tipc.h spawn.h util.h alloca.h endian.h \
-sys/endian.h)
+sys/endian.h sys/soundcard.h linux/soundcard.h)
 AC_HEADER_DIRENT
 AC_HEADER_MAJOR

diff -r ad855c779bf3 setup.py
--- a/setup.py  Thu Dec 03 17:32:05 2015 -0800
+++ b/setup.py  Sat Dec 05 14:36:21 2015 +0800
@@ -1550,7 +1547,9 @@
         # End multiprocessing
 
         # Platform-specific libraries
-        if host_platform.startswith(('linux', 'freebsd', 'gnukfreebsd')):
+        if (host_platform.startswith(('linux', 'freebsd', 'gnukfreebsd')) and
+            (sysconfig.get_config_var('HAVE_SYS_SOUNDCARD_H') or
+             sysconfig.get_config_var('HAVE_LINUX_SOUNDCARD_H'))):
             exts.append( Extension('ossaudiodev', ['ossaudiodev.c']) )
         else:
             missing.append('ossaudiodev')
diff -r ad855c779bf3 Modules/ossaudiodev.c
--- a/Modules/ossaudiodev.c     Thu Dec 03 17:32:05 2015 -0800
+++ b/Modules/ossaudiodev.c     Sat Dec 05 14:37:23 2015 +0800
@@ -31,7 +31,14 @@
 #endif
 
 #include <sys/ioctl.h>
-#include <sys/soundcard.h>
+
+#ifdef HAVE_SYS_SOUNDCARD_H
+# include <sys/soundcard.h>
+#elif defined(HAVE_LINUX_SOUNDCARD_H)
+# include <linux/soundcard.h>
+#else
+# error "Unable to find soundcard.h"
+#endif
 
 #if defined(linux)
