diff -ru bzip2-1.0.6/Makefile bzip2-1.0.6-p/Makefile
--- bzip2-1.0.6/Makefile	2010-09-10 22:46:02.000000000 +0000
+++ bzip2-1.0.6-p/Makefile	2014-01-20 01:35:53.252736910 +0000
@@ -15,16 +15,16 @@
 SHELL=/bin/sh
 
 # To assist in cross-compiling
-CC=gcc
-AR=ar
-RANLIB=ranlib
+CC?=gcc
+AR?=ar
+RANLIB?=ranlib
 LDFLAGS=
 
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS+=-Wall -Winline -O2 -g $(BIGFILES)
 
 # Where you want it installed when you do 'make install'
-PREFIX=/usr/local
+PREFIX?=/usr/local
 
 
 OBJS= blocksort.o  \
