--- util-linux-2.33.1/lib/fileutils.c.orig	2019-04-01 22:34:51.790851682 +0800
+++ util-linux-2.33.1/lib/fileutils.c	2019-04-01 22:35:38.840785664 +0800
@@ -54,8 +54,13 @@
 	/* Some use cases must be capable of being moved atomically
 	 * with rename(2), which is the reason why dir is here.  */
 	tmpenv = dir ? dir : getenv("TMPDIR");
-	if (!tmpenv)
+	if (!tmpenv) {
+#ifdef _PATH_TMP
 		tmpenv = _PATH_TMP;
+#else
+		return -1;
+#endif
+	}
 
 	rc = asprintf(&localtmp, "%s/%s.XXXXXX", tmpenv, prefix);
 	if (rc < 0)
