--- Python-3.5.1/Python/mysnprintf.c	2015-12-07 02:39:11.000000000 +0100
+++ Python-3.5-android/Python/mysnprintf.c	2017-01-21 14:47:33.095093000 +0100
@@ -1,5 +1,7 @@
 #include "Python.h"
 
+#include <android/log.h>
+
 /* snprintf() wrappers.  If the platform has vsnprintf, we use it, else we
    emulate it in a half-hearted way.  Even if the platform has it, we wrap
    it because platforms differ in what vsnprintf does in case the buffer
@@ -93,6 +95,9 @@
         assert(to_copy < size);
         memcpy(str, buffer, to_copy);
         str[to_copy] = '\0';
+
+	__android_log_print(ANDROID_LOG_DEBUG, "Python3", str);
+
     }
     PyMem_FREE(buffer);
 #endif

