--- gdbm-1.11/src/gdbmtool.h.orig	2016-04-07 14:08:27.732184893 +0800
+++ gdbm-1.11/src/gdbmtool.h	2016-04-07 14:08:56.858851195 +0800
@@ -239,6 +239,7 @@
 
 int variable_set (const char *name, int type, void *val);
 int variable_get (const char *name, int type, void **val);
+int variable_unset (const char *name);
 int variable_is_set (const char *name);
 int variable_is_true (const char *name);
 void variable_print_all (FILE *fp);
