--- gdbm-1.18.1/src/gdbmconst.h.orig	2019-03-29 22:30:12.890856233 +0800
+++ gdbm-1.18.1/src/gdbmconst.h	2019-03-29 22:30:30.264175151 +0800
@@ -55,4 +55,6 @@
 #define DEFAULT_CACHESIZE  100
 
 /* Maximum size representable by a size_t variable */
+#ifndef SIZE_T_MAX
 #define SIZE_T_MAX ((size_t)-1)
+#endif
--- gdbm-1.18.1/src/gdbmtool.c.orig	2019-03-29 22:58:05.888766018 +0800
+++ gdbm-1.18.1/src/gdbmtool.c	2019-03-29 22:58:15.682077046 +0800
@@ -38,8 +38,6 @@
 datum return_data;            /* Current data */
 int open_mode;                /* Default open mode */
 
-#define SIZE_T_MAX ((size_t)-1)
-
 unsigned input_line;
 
 
