diff -ru Python-3.3.3/Makefile.pre.in Python-3.3.3-android/Makefile.pre.in
--- Python-3.3.3/Makefile.pre.in	2013-11-17 07:22:55.000000000 +0000
+++ Python-3.3.3-android/Makefile.pre.in	2014-01-19 13:47:05.093593955 +0000
@@ -674,7 +674,7 @@
 $(GRAMMAR_H): $(GRAMMAR_INPUT) $(PGENSRCS)
 		@$(MKDIR_P) Include
 		$(MAKE) $(PGEN)
-		$(PGEN) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
+		$(HOSTPGEN) $(GRAMMAR_INPUT) $(GRAMMAR_H) $(GRAMMAR_C)
 $(GRAMMAR_C): $(GRAMMAR_H) $(GRAMMAR_INPUT) $(PGENSRCS)
 		$(MAKE) $(GRAMMAR_H)
 		touch $(GRAMMAR_C)
@@ -1240,6 +1240,7 @@
 # Install the dynamically loadable modules
 # This goes into $(exec_prefix)
 sharedinstall: sharedmods
+	CC='$(CC)' LDSHARED='$(BLDSHARED)' LDFLAGS='$(LDFLAGS)' OPT='$(OPT)' CROSS_COMPILE='$(CROSS_COMPILE)' \
 	$(RUNSHARED) $(PYTHON_FOR_BUILD) $(srcdir)/setup.py install \
 	   	--prefix=$(prefix) \
 		--install-scripts=$(BINDIR) \
diff -ru Python-3.3.3/configure Python-3.3.3-android/configure
--- Python-3.3.3/configure	2013-11-17 07:23:09.000000000 +0000
+++ Python-3.3.3-android/configure	2014-07-31 15:34:23.058883824 +0000
@@ -2943,13 +2943,18 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for python interpreter for cross build" >&5
 $as_echo_n "checking for python interpreter for cross build... " >&6; }
     if test -z "$PYTHON_FOR_BUILD"; then
-        for interp in python$PACKAGE_VERSION python3 python; do
-	    which $interp >/dev/null 2>&1 || continue
-	    if $interp -c 'import sys;sys.exit(not sys.version_info[:2] >= (3,3))'; then
-	        break
-	    fi
-            interp=
-	done
+        if test ! -z "$HOSTPYTHON" && "$HOSTPYTHON" -c 'import sys;sys.exit(not sys.version_info[:2] >= (3,3))'; then
+            interp="$HOSTPYTHON"
+        else
+            for interp in python$PACKAGE_VERSION python3 python; do
+	        which $interp >/dev/null 2>&1 || continue
+	        if $interp -c 'import sys;sys.exit(not sys.version_info[:2] >= (3,3))'; then
+	            break
+	        fi
+                interp=
+	    done
+        fi
+
         if test x$interp = x; then
 	    as_fn_error $? "python$PACKAGE_VERSION interpreter not found" "$LINENO" 5
 	fi
diff -ru Python-3.3.3/configure.ac Python-3.3.3-android/configure.ac
--- Python-3.3.3/configure.ac	2013-11-17 07:23:09.000000000 +0000
+++ Python-3.3.3-android/configure.ac	2014-07-31 15:31:35.846884194 +0000
@@ -56,13 +56,18 @@
 if test "$cross_compiling" = yes; then
     AC_MSG_CHECKING([for python interpreter for cross build])
     if test -z "$PYTHON_FOR_BUILD"; then
-        for interp in python$PACKAGE_VERSION python3 python; do
-	    which $interp >/dev/null 2>&1 || continue
-	    if $interp -c 'import sys;sys.exit(not sys.version_info@<:@:2@:>@ >= (3,3))'; then
-	        break
-	    fi
-            interp=
-	done
+        if test ! -z "$HOSTPYTHON" && "$HOSTPYTHON" -c 'import sys;sys.exit(not sys.version_info@<:@:2@:>@ >= (3,3))'; then
+            interp="$HOSTPYTHON"
+        else
+            for interp in python$PACKAGE_VERSION python3 python; do
+	        which $interp >/dev/null 2>&1 || continue
+	        if $interp -c 'import sys;sys.exit(not sys.version_info@<:@:2@:>@ >= (3,3))'; then
+	            break
+	        fi
+                interp=
+	    done
+        fi
+
         if test x$interp = x; then
 	    AC_MSG_ERROR([python$PACKAGE_VERSION interpreter not found])
 	fi

