diff -ru Python-3.3.3/Modules/pwdmodule.c Python-3.3.3-android/Modules/pwdmodule.c
--- Python-3.3.3/Modules/pwdmodule.c    2013-11-17 07:23:01.000000000 +0000
+++ Python-3.3.3-android/Modules/pwdmodule.c    2014-01-19 13:47:05.093593955 +0000
@@ -76,7 +76,7 @@
 #endif
     PyStructSequence_SET_ITEM(v, setIndex++, _PyLong_FromUid(p->pw_uid));
     PyStructSequence_SET_ITEM(v, setIndex++, _PyLong_FromGid(p->pw_gid));
-#ifdef __VMS
+#if defined(__VMS) || defined(__ANDROID__)
     SETS(setIndex++, "");
 #else
     SETS(setIndex++, p->pw_gecos);
diff -ru Python-3.3.3/Modules/_posixsubprocess.c Python-3.3.3-android/Modules/_posixsubprocess.c
--- Python-3.3.3/Modules/_posixsubprocess.c 2013-11-17 07:23:00.000000000 +0000
+++ Python-3.3.3-android/Modules/_posixsubprocess.c 2014-01-19 15:44:12.881751904 +0000
@@ -18,6 +18,12 @@
 #include <dirent.h>
 #endif
 
+#if defined(__ANDROID__)
+/* Android doesn't expose syscalls. Let's add the definition manually. */
+# include <sys/linux-syscalls.h>
+# define SYS_getdents64  __NR_getdents64
+#endif
+
 #if defined(sun)
 /* readdir64 is used to work around Solaris 9 bug 6395699. */
 # define readdir readdir64
diff -ru Python-3.3.3/Modules/socketmodule.c Python-3.3.3-android/Modules/socketmodule.c
--- Python-3.3.3/Modules/socketmodule.c        2013-11-17 07:23:02.000000000 +0000
+++ Python-3.3.3-android/Modules/socketmodule.c        2014-07-31 16:53:59.482873274 +0000
@@ -150,7 +150,7 @@
    On the other hand, not all Linux versions agree, so there the settings
    computed by the configure script are needed! */
 
-#ifndef linux
+#if !defined(linux) || __ANDROID__
 # undef HAVE_GETHOSTBYNAME_R_3_ARG
 # undef HAVE_GETHOSTBYNAME_R_5_ARG
 # undef HAVE_GETHOSTBYNAME_R_6_ARG
@@ -169,7 +169,7 @@
 #  define HAVE_GETHOSTBYNAME_R_3_ARG
 # elif defined(__sun) || defined(__sgi)
 #  define HAVE_GETHOSTBYNAME_R_5_ARG
-# elif defined(linux)
+# elif defined(linux) && !__ANDROID__
 /* Rely on the configure script */
 # else
 #  undef HAVE_GETHOSTBYNAME_R
diff -ru Python-3.3.3/Lib/subprocess.py Python-3.3.3-android/Lib/subprocess.py
--- Python-3.3.3/Lib/subprocess.py  2013-11-17 07:22:39.000000000 +0000
+++ Python-3.3.3-android/Lib/subprocess.py  2014-08-02 14:46:23.498143185 +0000
@@ -1343,9 +1343,13 @@
                 args = list(args)
 
             if shell:
-                args = ["/bin/sh", "-c"] + args
                 if executable:
-                    args[0] = executable
+                    main = executable
+                elif sys.platform == 'linux-android':
+                    main = '/system/bin/sh'
+                else:
+                    main = '/bin/sh'
+                args = [main, "-c"] + args
 
             if executable is None:
                 executable = args[0]
