diff -r 89116bd505cb Lib/platform.py
--- a/Lib/platform.py	Wed May 04 12:00:32 2016 -0400
+++ b/Lib/platform.py	Thu May 05 03:03:06 2016 +0800
@@ -690,6 +690,51 @@
 
     return release, vendor, vminfo, osinfo
 
+def _android_getprop(prop, default):
+    # On Android 4.1, `getprop` relies on an inherited file descriptor to
+    # gain access to the system property area. The id of this file descriptor
+    # is stored in the environment variable ANDROID_PROPERTY_WORKSPACE
+    pass_fds = ()
+    android_property_workspace = os.getenv('ANDROID_PROPERTY_WORKSPACE')
+    if android_property_workspace and ',' in android_property_workspace:
+        property_fd_str = android_property_workspace.split(',')[0]
+        if property_fd_str.isdigit():
+            pass_fds = (int(property_fd_str),)
+
+    try:
+        result = subprocess.check_output(['getprop', prop], pass_fds=pass_fds)
+        return result.decode('utf-8').strip()
+    except (OSError, subprocess.CalledProcessError, UnicodeDecodeError):
+        return default
+
+def android_ver(version='', sdk='', versioninfo=('', ''), hardwareinfo=('', '')):
+    """ Attempt to get the Android version information.
+
+        The function checks Android system properties to retrieve info.
+
+        Returns a (version, sdk, versioninfo, hardwareinfo) tuple with
+        versioninfo being a (codename, incremental) tuple and hardwareinfo
+        being a (boardname, abi) tuple.
+
+        Values which cannot be determined are set to the defaults
+        given as parameters (which all default to '').
+    """
+
+    version = _android_getprop('ro.build.version.release', version)
+    sdk = _android_getprop('ro.build.version.sdk', sdk)
+
+    codename, incremental = versioninfo
+    codename = _android_getprop('ro.build.version.codename', codename)
+    incremental = _android_getprop('ro.build.version.incremental', incremental)
+    versioninfo = codename, incremental
+
+    boardname, abi = hardwareinfo
+    boardname = _android_getprop('ro.product.board', boardname)
+    abi = _android_getprop('ro.product.cpu.abi', abi)
+    hardwareinfo = boardname, abi
+
+    return version, sdk, versioninfo, hardwareinfo
+
 ### System name aliasing
 
 def system_alias(system, release, version):
diff -ru Python-3.3.5/Lib/subprocess.py Python-3.3.5-android/Lib/subprocess.py
--- Python-3.3.5/Lib/subprocess.py	2014-03-09 09:40:13.000000000 +0100
+++ Python-3.3.5-android/Lib/subprocess.py	2014-08-04 22:19:36.000000000 +0200
@@ -1343,9 +1343,18 @@
                 args = list(args)

             if shell:
-                args = ["/bin/sh", "-c"] + args
                 if executable:
-                    args[0] = executable
+                    main = executable
+                elif os.path.isfile('/bin/sh'):
+                    main = '/bin/sh'
+                else:
+                    import platform
+                    if platform.android_ver()[0]:
+                        main = '/system/bin/sh'
+                    else:
+                        raise RuntimeError('Could not find system shell')
+
+                args = [main, "-c"] + args

             if executable is None:
                 executable = args[0]
diff -ru Python-3.3.5/Lib/test/test_subprocess.py Python-3.3.5-android/Lib/test/test_subprocess.py
--- Python-3.3.5/Lib/test/test_subprocess.py	2014-03-09 09:40:19.000000000 +0100
+++ Python-3.3.5-android/Lib/test/test_subprocess.py	2014-08-04 22:19:36.000000000 +0200
@@ -17,6 +17,7 @@
 import shutil
 import gc
 import textwrap
+import platform

 try:
     import resource
@@ -1356,7 +1357,10 @@
         fd, fname = mkstemp()
         # reopen in text mode
         with open(fd, "w", errors="surrogateescape") as fobj:
-            fobj.write("#!/bin/sh\n")
+            if platform.android_ver()[0]:
+                fobj.write('#!/system/bin/sh\n')
+            else:
+                fobj.write("#!/bin/sh\n")
             fobj.write("exec '%s' -c 'import sys; sys.exit(47)'\n" %
                        sys.executable)
         os.chmod(fname, 0o700)
@@ -1401,7 +1405,10 @@
         fd, fname = mkstemp()
         # reopen in text mode
         with open(fd, "w", errors="surrogateescape") as fobj:
-            fobj.write("#!/bin/sh\n")
+            if platform.android_ver()[0]:
+                fobj.write('#!/system/bin/sh\n')
+            else:
+                fobj.write("#!/bin/sh\n")
             fobj.write("exec '%s' -c 'import sys; sys.exit(47)'\n" %
                        sys.executable)
         os.chmod(fname, 0o700)
diff -ru  cpython-fac5562ae9c6/configure.ac.orig cpython-fac5562ae9c6/configure.ac
--- cpython-fac5562ae9c6/configure.ac.orig	2015-11-24 14:19:12.000000000 +0800
+++ cpython-fac5562ae9c6/configure.ac	2015-11-24 16:39:35.356301199 +0800
@@ -858,6 +858,21 @@
 AC_SUBST(PLATDIR)
 AC_SUBST(PLATFORM_TRIPLET)
 
+# Test if we're running on Android.
+AC_MSG_CHECKING(if target is Android-based)
+AC_EGREP_CPP(yes,
+[
+#if __ANDROID__
+yes
+#endif
+], [
+    AC_MSG_RESULT(yes)
+    with_android=yes
+   ], [
+    AC_MSG_RESULT(no)
+    with_android=no
+   ]
+)
 
 AC_MSG_CHECKING([for -Wl,--no-as-needed])
 save_LDFLAGS="$LDFLAGS"
@@ -1109,7 +1094,14 @@
 	  LDLIBRARY='libpython$(LDVERSION).so'
 	  BLDLIBRARY='-L. -lpython$(LDVERSION)'
	  RUNSHARED=LD_LIBRARY_PATH=`pwd`${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
-	  INSTSONAME="$LDLIBRARY".$SOVERSION
+
+	  if test "$with_android" != yes
+	  then
+		INSTSONAME="$LDLIBRARY".$SOVERSION
+	  else
+		INSTSONAME="$LDLIBRARY"
+	  fi
+
 	  if test "$with_pydebug" != yes
           then
 	      PY3LIBRARY=libpython3.so

diff -ru Python-3.4.2/Makefile.pre.in Python-3.4.2-android/Makefile.pre.in
--- Python-3.4.2/Makefile.pre.in	2015-03-04 16:25:36.000000000 +0100
+++ Python-3.4.2-android/Makefile.pre.in	2015-03-04 16:27:27.000000000 +0100
@@ -568,7 +568,7 @@
 	    *\ -s*|s*) quiet="-q";; \
 	    *) quiet="";; \
 	esac; \
-	$(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED)' OPT='$(OPT)' \
+	$(RUNSHARED) CC='$(CC)' LDSHARED='$(BLDSHARED) -lpython$(LDVERSION)' OPT='$(OPT)' \
 		_TCLTK_INCLUDES='$(TCLTK_INCLUDES)' _TCLTK_LIBS='$(TCLTK_LIBS)' \
 		$(PYTHON_FOR_BUILD) $(srcdir)/setup.py $$quiet build
 
diff -Nru Python-3.4.2/Makefile.pre.in Python-3.4.2-android/Makefile.pre.in
--- Python-3.4.2/Makefile.pre.in	2015-06-27 17:04:23.885777456 +0000
+++ Python-3.4.2-android/Makefile.pre.in	2015-06-27 17:05:27.709777315 +0000
@@ -585,11 +585,9 @@
 	$(RANLIB) $@
 
 libpython$(LDVERSION).so: $(LIBRARY_OBJS)
+	$(BLDSHARED) -Wl,-h$(INSTSONAME) -o $(INSTSONAME) $(LIBRARY_OBJS) $(MODLIBS) $(SHLIBS) $(LIBC) $(LIBM) $(LDLAST); \
 	if test $(INSTSONAME) != $(LDLIBRARY); then \
-		$(BLDSHARED) -Wl,-h$(INSTSONAME) -o $(INSTSONAME) $(LIBRARY_OBJS) $(MODLIBS) $(SHLIBS) $(LIBC) $(LIBM) $(LDLAST); \
 		$(LN) -f $(INSTSONAME) $@; \
-	else \
-		$(BLDSHARED) -o $@ $(LIBRARY_OBJS) $(MODLIBS) $(SHLIBS) $(LIBC) $(LIBM) $(LDLAST); \
 	fi
 
 libpython3.so:	libpython$(LDVERSION).so
