diff --git a/setup.py b/setup.py
index 3b3d097454..418c4864a1 100644
--- a/setup.py
+++ b/setup.py
@@ -262,8 +262,8 @@ class PyBuildExt(build_ext):
         # unfortunately, distutils doesn't let us provide separate C and C++
         # compilers
         if compiler is not None:
-            (ccshared,cflags) = sysconfig.get_config_vars('CCSHARED','CFLAGS')
-            args['compiler_so'] = compiler + ' ' + ccshared + ' ' + cflags
+            (ccshared,cflags, cppflags) = sysconfig.get_config_vars('CCSHARED','CFLAGS', 'CPPFLAGS')
+            args['compiler_so'] = compiler + ' ' + ccshared + ' ' + cppflags + ' ' + cflags
         self.compiler.set_executables(**args)
 
         build_ext.build_extensions(self)
