diff -ru cpython-fac5562ae9c6/Makefile.pre.in.orig cpython-fac5562ae9c6/Makefile.pre.in
--- cpython-fac5562ae9c6/Makefile.pre.in.orig	2015-11-26 00:01:34.116741659 +0800
+++ cpython-fac5562ae9c6/Makefile.pre.in	2015-11-26 00:04:05.046746704 +0800
@@ -1416,6 +1416,7 @@
 # Install the dynamically loadable modules
 # This goes into $(exec_prefix)
 sharedinstall: sharedmods
+	CC='$(CC)' LDSHARED='$(BLDSHARED)' LDFLAGS='$(LDFLAGS)' OPT='$(OPT)' CROSS_COMPILE='$(CROSS_COMPILE)' \
 	$(RUNSHARED) $(PYTHON_FOR_BUILD) $(srcdir)/setup.py install \
 	   	--prefix=$(prefix) \
 		--install-scripts=$(BINDIR) \
