diff -ru cpython-fac5562ae9c6/Makefile.pre.in.orig cpython-fac5562ae9c6/Makefile.pre.in
--- cpython-fac5562ae9c6/Makefile.pre.in.orig	2015-11-26 00:01:34.116741659 +0800
+++ cpython-fac5562ae9c6/Makefile.pre.in	2015-11-26 00:04:05.046746704 +0800
@@ -1416,6 +1416,7 @@
 # Install the dynamically loadable modules
 # This goes into $(exec_prefix)
 sharedinstall: sharedmods
+	CC='$(CC)' LDSHARED='$(BLDSHARED)' LDFLAGS='$(LDFLAGS)' OPT='$(OPT)' CROSS_COMPILE='$(CROSS_COMPILE)' \
 	$(RUNSHARED) $(PYTHON_FOR_BUILD) $(srcdir)/setup.py install \
 	   	--prefix=$(prefix) \
 		--install-scripts=$(BINDIR) \
diff -r 856d50130154 setup.py
--- a/setup.py	Tue Apr 05 10:17:38 2016 +0000
+++ b/setup.py	Wed Apr 06 02:05:20 2016 +0800
@@ -492,6 +492,10 @@
             self.add_gcc_paths()
         self.add_multiarch_paths()
 
+        if 'SYSROOT' in os.environ:
+            add_dir_to_list(self.compiler.include_dirs, os.path.join(os.environ['SYSROOT'], 'usr/include'))
+            add_dir_to_list(self.compiler.library_dirs, os.path.join(os.environ['SYSROOT'], 'usr/lib'))
+
         # Add paths specified in the environment variables LDFLAGS and
         # CPPFLAGS for header and library files.
         # We must get the values from the Makefile and not the environment
