diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -2008,17 +2011,17 @@ class PyBuildExt(build_ext):
             sources = ['_decimal/_decimal.c']
             depends = ['_decimal/docstrings.h']
         else:
             srcdir = sysconfig.get_config_var('srcdir')
             include_dirs = [os.path.abspath(os.path.join(srcdir,
                                                          'Modules',
                                                          '_decimal',
                                                          'libmpdec'))]
-            libraries = []
+            libraries = self.detect_math_libs()
             sources = [
               '_decimal/_decimal.c',
               '_decimal/libmpdec/basearith.c',
               '_decimal/libmpdec/constants.c',
               '_decimal/libmpdec/context.c',
               '_decimal/libmpdec/convolute.c',
               '_decimal/libmpdec/crt.c',
               '_decimal/libmpdec/difradix2.c',
