--- a/setup.py	2017-02-04 14:54:29.334671104 +0800
+++ b/setup.py	2017-02-04 14:56:03.731337064 +0800
@@ -508,7 +508,8 @@
         for env_var, arg_name, dir_list in (
                 ('LDFLAGS', '-R', self.compiler.runtime_library_dirs),
                 ('LDFLAGS', '-L', self.compiler.library_dirs),
-                ('CPPFLAGS', '-I', self.compiler.include_dirs)):
+                ('CPPFLAGS', '-I', self.compiler.include_dirs),
+                ('CPPFLAGS', '-D', self.compiler.macros)):
             env_val = sysconfig.get_config_var(env_var)
             if env_val:
                 parser = argparse.ArgumentParser()
@@ -516,8 +517,14 @@
                 parser.add_argument(arg_name, dest="dirs", action="append")
                 options = parser.parse_args(env_val.split())
                 if options.dirs:
-                    for directory in reversed(options.dirs):
-                        add_dir_to_list(dir_list, directory)
+                    if arg_name == '-D':
+                        for macro in options.dirs:
+                            macro_pair = macro.split('=', maxsplit=1)
+                            self.compiler.macros.append((macro_pair[0],
+                                                         macro_pair[1] if len(macro_pair) > 1 else None))
+                    else:
+                        for directory in reversed(options.dirs):
+                            add_dir_to_list(dir_list, directory)
 
         if (not cross_compiling and
                 os.path.normpath(sys.base_prefix) != '/usr' and
