diff -r 8b85df50d906 configure.ac
--- a/configure.ac	Sun Mar 20 12:51:16 2016 +0200
+++ b/configure.ac	Mon Mar 21 01:32:59 2016 +0800
@@ -3996,6 +4033,25 @@
 fi],
 [AC_MSG_RESULT(default LIBC="$LIBC")])
 
+# check for --with-android-support=...
+AC_SUBST(WITH_ANDROID_SUPPORT)
+AC_MSG_CHECKING(for --with-android-support=STRING)
+AC_ARG_WITH(android-support,
+            AS_HELP_STRING([--with(out)-android-support], [using android-support library from Android NDK]))
+
+if test -z "$with_android_support"
+then
+    with_android_support="no"
+fi
+if test "$with_android_support" != "no"
+then
+    AC_DEFINE(WITH_ANDROID_SUPPORT, 1, 
+     [Define if you want to use android-support package from Android NDK])
+    LIBC="${LIBC} -landroid_support -lm"
+    CPPFLAGS="-I${prefix}/include/android_support ${CPPFLAGS}"
+fi
+AC_MSG_RESULT($with_android_support)
+
 # **************************************
 # * Check for gcc x64 inline assembler *
 # **************************************
diff -r 8b85df50d906 setup.py
--- a/setup.py	Sun Mar 20 12:51:16 2016 +0200
+++ b/setup.py	Mon Mar 21 01:15:51 2016 +0800
@@ -1331,8 +1328,12 @@
             curses_defines.append(('HAVE_NCURSESW', '1'))
             curses_defines.append(('_XOPEN_SOURCE_EXTENDED', '1'))
 
+        android_support_libs = []
+        if sysconfig.get_config_var('WITH_ANDROID_SUPPORT'):
+            android_support_libs.append('android_support')
+
         if curses_library.startswith('ncurses'):
-            curses_libs = [curses_library]
+            curses_libs = [curses_library] + android_support_libs
             exts.append( Extension('_curses', ['_cursesmodule.c'],
                                    include_dirs=curses_includes,
                                    define_macros=curses_defines,
@@ -1347,6 +1348,7 @@
             else:
                 curses_libs = ['curses']
 
+            curses_libs.extend(android_support_libs)
             exts.append( Extension('_curses', ['_cursesmodule.c'],
                                    define_macros=curses_defines,
                                    libraries = curses_libs) )
diff -r 8b85df50d906 Modules/_cursesmodule.c
--- a/Modules/_cursesmodule.c	Sun Mar 20 12:51:16 2016 +0200
+++ b/Modules/_cursesmodule.c	Sun Mar 20 22:23:10 2016 +0800
@@ -122,6 +122,11 @@
 #endif
 
 #ifdef HAVE_LANGINFO_H
+/*
+ * In NDK android-support package langinfo.h, LC_CTYPE is referenced but
+ * locale.h not included
+ * */
+#include <locale.h>
 #include <langinfo.h>
 #endif
 
