diff --git a/setup.py b/setup.py
index 3b3d097454..2fd04e7b82 100644
--- a/setup.py
+++ b/setup.py
@@ -254,18 +254,6 @@ class PyBuildExt(build_ext):
             self.extensions = [x for x in self.extensions if x not in
                                removed_modules]
 
-        # When you run "make CC=altcc" or something similar, you really want
-        # those environment variables passed into the setup.py phase.  Here's
-        # a small set of useful ones.
-        compiler = os.environ.get('CC')
-        args = {}
-        # unfortunately, distutils doesn't let us provide separate C and C++
-        # compilers
-        if compiler is not None:
-            (ccshared,cflags) = sysconfig.get_config_vars('CCSHARED','CFLAGS')
-            args['compiler_so'] = compiler + ' ' + ccshared + ' ' + cflags
-        self.compiler.set_executables(**args)
-
         build_ext.build_extensions(self)
 
         for ext in self.extensions:
