--- openssl/crypto/bn/bn_lcl.h.orig	2016-09-10 16:29:01.975108970 +0800
+++ openssl/crypto/bn/bn_lcl.h	2016-09-10 16:29:26.578441593 +0800
@@ -428,7 +428,7 @@
 #   endif
 #  elif defined(__mips) && (defined(SIXTY_FOUR_BIT) || defined(SIXTY_FOUR_BIT_LONG))
 #   if defined(__GNUC__) && __GNUC__>=2
-#    if __GNUC__>4 || (__GNUC__>=4 && __GNUC_MINOR__>=4)
+#    if __GNUC__>4 || (__GNUC__>=4 && __GNUC_MINOR__>=4) || defined(__clang__)
                                      /* "h" constraint is no more since 4.4 */
 #     define BN_UMULT_HIGH(a,b)          (((__uint128_t)(a)*(b))>>64)
 #     define BN_UMULT_LOHI(low,high,a,b) ({     \
