diff --git a/Configurations/10-main.conf b/Configurations/10-main.conf
index 59234a8a5..51ee7c8e0 100644
--- a/Configurations/10-main.conf
+++ b/Configurations/10-main.conf
@@ -904,12 +904,12 @@ sub vms_info {
         # systems are perfectly capable of executing binaries targeting
         # Froyo. Keep in mind that in the nutshell Android builds are
         # about JNI, i.e. shared libraries, not applications.
-        cflags           => add(picker(default => "-mandroid -fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
-        bin_cflags       => "-pie",
+        cflags           => add(picker(default => "-fno-integrated-as -D__ANDROID_API__=\$(ANDROID_API_LEVEL) -fPIC --sysroot=\$(UNIFIED_SYSROOT) -Wa,--noexecstack")),
+        bin_cflags       => "-pie --sysroot=\$(ARCH_SYSROOT)",
     },
     "android-x86" => {
         inherit_from     => [ "android", asm("x86_asm") ],
-        cflags           => add(picker(release => "-fomit-frame-pointer")),
+        cflags           => add(picker(release => "-I\$(UNIFIED_SYSROOT)/include/i686-linux-android -fomit-frame-pointer")),
         bn_ops           => "BN_LLONG",
         perlasm_scheme   => "android",
     },
@@ -936,23 +936,27 @@ sub vms_info {
     #
     "android-armeabi" => {
         inherit_from     => [ "android", asm("armv4_asm") ],
+        cflags           => add(picker(default => "-I\$(UNIFIED_SYSROOT)/include/arm-linux-androideabi")),
     },
     "android-mips" => {
         inherit_from     => [ "android", asm("mips32_asm") ],
+        cflags           => add(picker(default => "-I\$(UNIFIED_SYSROOT)/include/mipsel-linux-android")),
         perlasm_scheme   => "o32",
     },
 
     "android64" => {
         inherit_from     => [ "linux-generic64" ],
-        cflags           => add(picker(default => "-mandroid -fPIC --sysroot=\$(CROSS_SYSROOT) -Wa,--noexecstack")),
-        bin_cflags       => "-pie",
+        cflags           => add(picker(default => "-fno-integrated-as -D__ANDROID_API__=\$(ANDROID_API_LEVEL) -fPIC --sysroot=\$(UNIFIED_SYSROOT) -Wa,--noexecstack")),
+        bin_cflags       => "-pie --sysroot=\$(ARCH_SYSROOT)",
     },
     "android64-aarch64" => {
         inherit_from     => [ "android64", asm("aarch64_asm") ],
+        cflags           => add(picker(default => "-I\$(UNIFIED_SYSROOT)/include/aarch64-linux-android")),
         perlasm_scheme   => "linux64",
     },
     "android64-x86_64" => {
         inherit_from     => [ "android64", asm("x86_64_asm") ],
+        cflags           => add(picker(default => "-I\$(UNIFIED_SYSROOT)/include/x86_64-linux-android")),
         perlasm_scheme   => "elf",
     },
     "android64-mips64" => {
@@ -966,6 +970,7 @@ sub vms_info {
         # prior versions original MIPS binary code will fail.
         #
         inherit_from     => [ "android64", asm("mips64_asm") ],
+        cflags           => add(picker(default => "-I\$(UNIFIED_SYSROOT)/include/mips64el-linux-android")),
         perlasm_scheme   => "64",
     },
 
