diff --git a/Configure b/Configure
index b2e2e4787..3fc82b643 100755
--- a/Configure
+++ b/Configure
@@ -970,6 +970,7 @@ $config{cross_compile_prefix} = $ENV{'CROSS_COMPILE'}
 $config{perl} =    ($^O ne "VMS" ? $^X : "perl");
 $config{hashbangperl} =
     $ENV{'HASHBANGPERL'}           || $ENV{'PERL'}     || "/usr/bin/env perl";
+$config{hashbangsh} = $ENV{'HASHBANGSH'} || "/bin/sh";
 $target{cc} =      $ENV{'CC'}      || $target{cc}      || "cc";
 $target{cxx} =     $ENV{'CXX'}     || $target{cxx}     || "c++";
 $target{ranlib} =  $ENV{'RANLIB'}  || $target{ranlib}  ||
diff --git a/util/shlib_wrap.sh.in b/util/shlib_wrap.sh.in
index 27a1e513c..a5fdbf192 100755
--- a/util/shlib_wrap.sh.in
+++ b/util/shlib_wrap.sh.in
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!{- $config{hashbangsh} -}
 
 [ $# -ne 0 ] || set -x		# debug mode without arguments:-)
 
