--- readline-6.3/support/shlib-install.orig	2015-12-03 02:44:23.573947502 +0800
+++ readline-6.3/support/shlib-install	2015-12-03 02:45:00.480615403 +0800
@@ -118,6 +118,9 @@
 # Create symlinks to the installed library.  This section is incomplete.
 #
 case "$host_os-$host_vendor" in
+*android*)
+	# Do nothing on Android
+	;;
 *linux*|freebsd*-gentoo)
 	# libname.so.M -> libname.so.M.N
 	${echo} ${RM} ${INSTALLDIR}/$LINK2
--- readline-6.3/support/shobj-conf.orig	2015-12-03 02:54:40.873968133 +0800
+++ readline-6.3/support/shobj-conf	2015-12-03 02:54:47.033968338 +0800
@@ -122,6 +122,17 @@
 	SHLIB_LIBVERSION='$(SHLIB_LIBSUFF).$(SHLIB_MAJOR)'
 	;;
 
+# Android, similar to Linux but without versioned .so
+*android*)
+	SHOBJ_CFLAGS=-fPIC
+	SHOBJ_LD='${CC}'
+	SHOBJ_LDFLAGS='-shared -Wl,-soname,$@'
+
+	SHLIB_XLDFLAGS='-Wl,-rpath,$(libdir) -Wl,-soname,$@ -Wl,--no-undefined'
+	SHLIB_LIBVERSION='$(SHLIB_LIBSUFF)'
+	SHLIB_LIBS='-lncursesw'
+	;;
+
 # All versions of Linux (including Gentoo/FreeBSD) or the semi-mythical GNU Hurd.
 linux*-*|gnu*-*|k*bsd*-gnu-*|freebsd*-gentoo)
 	SHOBJ_CFLAGS=-fPIC
